File: leetcode/easy/0013_roman_to_integer/solution.cpp
#include <string>
#include <unordered_map>

int romanToInt(const std::string& s) {
    static const std::unordered_map<char, int> values = {
        {'I', 1}, {'V', 5}, {'X', 10},
        {'L', 50}, {'C', 100},
        {'D', 500}, {'M', 1000}
    };

    int total = 0;

    for (size_t i = 0; i < s.size(); ++i) {
        int curr = values.at(s[i]);

        if (i + 1 < s.size() && curr < values.at(s[i + 1])) {
            total -= curr;
        } else {
            total += curr;
        }
    }

    return total;
}

File: leetcode/easy/0013_roman_to_integer/solution.py
def romanToInt(s: str) -> int:
    roman_to_int = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    }

    result = 0
    for curr, next in zip(s, s[1:]):
        result += - roman_to_int[curr] if roman_to_int[curr] < roman_to_int[next] else roman_to_int[curr]
    return result + roman_to_int[s[-1]]
